### PR TITLE
Fix/mnt 20916 log4j configuration hierarchy

### PR DESCRIPTION
**Pull request by nithinnambiar**
Updated Log4JHierarchyInit  to merge the log4j properties and initialise the config only once.

### CODE REVIEW

File: src/main/java/org/alfresco/repo/admin/Log4JHierarchyInit.java [hunk 1]
```diff
@@ -145,8 +155,10 @@ private void importLogSettings(Method method, String springUrl)
         {
             try
             {
-                URL url = resource.getURL();
-                method.invoke(null, url);
+                InputStream inputStream=resource.getInputStream();
```
Comment (alandavis) [inline src/main/java/org/alfresco/repo/admin/Log4JHierarchyInit.java:158]: spaces either side of =

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java [hunk 2]
```diff
@@ -63,4 +73,38 @@ public void testAddingLog4jProperties() throws Throwable
         // We expect DEBUG to be on
         assertTrue("DEBUG was not enabled for logger " + this.getClass(), log.isDebugEnabled());
     }
+
+    public void setUpStreams() throws UnsupportedEncodingException
+    {
+        sysErr = System.err;
+
+        System.setErr(new PrintStream(errContent,false, "UTF-8"));
+
```
Comment (alandavis) [inline src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java:82]: remove blank lines at the end of methods

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java [hunk 3]
```diff
@@ -63,4 +73,38 @@ public void testAddingLog4jProperties() throws Throwable
...
+    {
+        sysErr = System.err;
+
+        System.setErr(new PrintStream(errContent,false, "UTF-8"));
+
+    }
+
+
+    public void revertStreams() {
+        System.setErr(sysErr);
```
Comment (alandavis) [inline src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java:87]: Balance {
}
so they are
{
}
not K&K style.

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java [hunk 4]
```diff
@@ -63,4 +73,38 @@ public void testAddingLog4jProperties() throws Throwable
...
+
```
Comment (alandavis) [inline src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java:88]: more blank lines at the end of a method

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java [hunk 5]
```diff
@@ -63,4 +73,38 @@ public void testAddingLog4jProperties() throws Throwable
...
+    }
+
+    public void testLog4jAppenderClosedError() throws Throwable
+    {
+        setUpStreams();
+        Log log = LogFactory.getLog(this.getClass());
+
+        Log4JHierarchyInit log4JHierarchyInit =(Log4JHierarchyInit)ctx.getBean("log4JHierarchyInit");
+
+        Log log2= LogFactory.getLog(FileFolderPerformanceTester.class);
```
Comment (alandavis) [inline src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java:98]: need space before =

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java [hunk 6]
```diff
@@ -63,4 +73,38 @@ public void testAddingLog4jProperties() throws Throwable
...
+        log4JHierarchyInit.init();
+
+        log2.info("test");
+
+        // We expect DEBUG to be on
+        assertTrue("DEBUG was not enabled for logger " + this.getClass(), log.isDebugEnabled());
+        assertFalse(errContent.toString().contains("Attempted to append to closed appender named"));
+
+        revertStreams();
+
```
Comment (alandavis) [inline src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java:108]: remove blank line

File: src/main/java/org/alfresco/repo/admin/Log4JHierarchyInit.java (hunk 1 above)
Comment (nithinnambiar) [reply src/main/java/org/alfresco/repo/admin/Log4JHierarchyInit.java]: Done

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java (hunk 2 above)
Comment (nithinnambiar) [reply src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java]: Done

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java (hunk 3 above)
Comment (nithinnambiar) [reply src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java:85]: Done.

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java (hunk 5 above)
Comment (nithinnambiar) [reply src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java]: Done

File: src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java (hunk 6 above)
Comment (nithinnambiar) [reply src/test/java/org/alfresco/repo/admin/Log4JHierarchyInitTest.java:105]: Done

**Pull request merged**